docs/design/003-wasm-support.md

```diff
@@ -0,0 +1,69 @@
+# Wasm Support in Inspektor Gadget
+
+We want to further extend the flexibility of [image-based
+gadgets](./002-containerized-gadgets.md) by allowing users to run custom code in
+user space.
+
+## Wasm Capabilities
+
+This section covers the features Wasm should provide to gadget developers.
+
+### Manipulate Data Sources
+
+The Wasm module should be able to manipulate the data coming from other sources,
+(eBPF being one of them). It'll allow the Wasm module to change the format of
+the fields, for instance, performing string manipulations that are difficult to
+implement in eBPF, to enrich events with new information it gathers from other
+places, etc. The Wasm program should also be able to create a new data source on
+its own to provide new information.
+
+This is the list of operations the Wasm module should be able to make:
+
+- Change the value of fields
+  - Implement data manipulations impossible to make in eBPF
+- Add fields
+  - Create "virtual fields" by combining existing ones. Like the "call" field on the trace mount gadget
+  - Create new fields to enrich the event with information the Wasm module has
+- Change fields attributes
+  - Hide / show the field
+  - Change the width used to print it
```

Comment:
"add annotations" might be another important one (basically also what is used for width)